pkg/kvfile/kvfile.go

```diff
@@ -0,0 +1,130 @@
+// Package kvfile provides utilities to parse line-delimited key/value files
+// such as used for label-files and env-files.
+//
+// # File format
+//
+// key/value files use the following syntax:
+//
+//   - File must be valid UTF-8.
+//   - BOM headers are removed.
+//   - Leading whitespace is removed for each line.
+//   - Lines starting with "#" are ignored.
+//   - Empty lines are ignored.
+//   - Key/Value pairs are provided as "KEY[=<VALUE>]".
+//   - Maximum line-length is limited to [bufio.MaxScanTokenSize].
+//
+// # Interpolation, substitution, and escaping
+//
+// Both keys and values are used as-is; no interpolation, substitution or
+// escaping is supported, and quotes are considered part of the key or value.
+// Whitespace in values (including leading and trailing) is preserved. Given
+// that the file format is line-delimited, neither key, nor value, can contain
+// newlines.
+//
+// # Key/Value pairs
+//
+// Key/Value pairs take the following format:
+//
+//	KEY[=<VALUE>]
+//
+// KEY is required and may not contain whitespaces or NUL characters. Any
+// other character (except for the "=" delimiter) are accepted, but  it is
+// recommended to use a subset of the POSIX portable character set, as
+// outlined in [Environment Variables].
```

Comment:
Indeed, I remember having a discussion about this some time ago and while **some programs** will only accept a subset of the portable character set, there's no guarantees and often applications that people run _rely on_ other random characters, so we shouldn't do much validating here ourselves.

Reply:
Yeah, compose recently ran into some of these. And, yes, Linux accepts "just about anything" (anything except `NUL` terminator). And, just because it _does_ doesn't mean you _should_ ... but some systems do, which is why we don't do validation on any of these.

Reply:
It might've been on Compose, yeah 😅